qsc/grad_B_tensor.py

```diff
@@ -1177,3 +1190,233 @@ def calculate_grad_grad_B_tensor(self, two_ways=False):
     grad_grad_B_alt[:,2,2,2] =(-2*B0*curvature*curvature)/sign_G
 
     self.grad_grad_B_alt = grad_grad_B_alt
+
+
+def Bfield_cylindrical(self, r=0, theta=0):
+    '''
+    Function to calculate the magnetic field vector B=(B_R,B_phi,B_Z) at
+    every point along the axis (hence with nphi points) where R, phi and Z
+    are the standard cylindrical coordinates for a given
+    near-axis radius r and a Boozer poloidal angle vartheta (not theta).
+
+    Args:
+      r: the near-axis radius
+      theta: the Boozer poloidal angle vartheta (= theta-N*phi)
+    '''
+    
+    # Define auxiliary variables
+    t = self.tangent_cylindrical.transpose()
+    n = self.normal_cylindrical.transpose()
+    b = self.binormal_cylindrical.transpose()
+    B0 = self.B0
+    sG = self.sG
+    G0 = self.G0
+    X1c = self.X1c
+    X1s = self.X1s
+    Y1c = self.Y1c
+    Y1s = self.Y1s
+    d_l_d_varphi = self.d_l_d_varphi
+    curvature = self.curvature
+    torsion = self.torsion
+    iotaN = self.iotaN
+    d_X1c_d_varphi = self.d_X1c_d_varphi
+    d_X1s_d_varphi = self.d_X1s_d_varphi
+    d_Y1s_d_varphi = self.d_Y1s_d_varphi
+    d_Y1c_d_varphi = self.d_Y1c_d_varphi
+
+    B0_vector = sG * B0 * t
+
+    if r == 0:
+        return B0_vector
+    else:
+        factor = B0*B0/G0
```

Comment:
How about adding a comment referencing eq (3.6) in the figures of merit paper for these formulae.